forum/qa-content/qa-page.js

```diff
@@ -870,3 +876,95 @@ function qa_ajax_error()
     }
 
 } ( document ) );
+
+
+/**
+ *	Add button, which will let users to copy link of topic/comment they are currently reading and share it
+ */
+;( function( document ) {
+
+	'use strict';
+
+	var copyToClipboard = window.initializeCopyToClipboard;
+
+    console.info( 'copyToClipboard: ', copyToClipboard, ' /boolean: ' , !!copyToClipboard );
+
+	if ( !!copyToClipboard ) {
+		var shareMethodPopup = ( function() {
+			var popup = document.createElement( 'div' );
+			popup.classList.add( 'share-popup' );
+            popup.id = 'share-popup';
+
+			var shareLinkField = document.createElement( 'input' );
+			shareLinkField.type = 'text';
+			shareLinkField.readonly = true;
+            shareLinkField.id = 'shareLinkField';
+			shareLinkField.value = '';
+
+			var copyLinkBtn = document.createElement( 'button' );
+			copyLinkBtn.type = 'button';
+            copyLinkBtn.textContent = 'Kopiuj link';
+            copyLinkBtn.classList.add( 'copy-shared-link' );
+
+			popup.appendChild( shareLinkField );
+			popup.appendChild( copyLinkBtn );
+
+			return {
+                popup: popup,
+                shareLinkField: shareLinkField
+            };
+		}() );
+
+		var addShareButton = function() {
+
+			var topicActionButtonsBar = document.querySelector( '.qa-q-view-buttons' );
+			var answersActionButtonsBar = Array.from( document.querySelectorAll( '.qa-a-item-buttons' ) );
+			var commentsActionButtonsBar = Array.from( document.querySelectorAll( '.qa-c-item-buttons' ) );
+
+			var allActionButtonsBar = answersActionButtonsBar.concat( commentsActionButtonsBar );
+			allActionButtonsBar.push( topicActionButtonsBar );
+
+			for ( var i = 0, len = allActionButtonsBar.length; i < len; i++ ) {

```

Comment:
`len = allActionButtonsBar.length;` - what is the points of assignment if you don't use `len` inside the for loop?
 
`for ( var i = 0; i < allActionButtonsBar.length; i++ ) { (...) }`